src/util.rs

```diff
@@ -12,3 +12,45 @@ macro_rules! evaluate {
         results
     };
 }
+
+// parse object of type type_name from vec<u8> mess
+#[macro_export]
+macro_rules! get_message {
+    ($type_name:ty, $mess:expr) => {
+        protobuf::parse_from_bytes::<$type_name>($mess)
+    };
+}
+
+#[macro_export]
+macro_rules! poll_events {
+    ($s:expr, $ret:expr) => {
+        match $s.get_events() {
+            Err(zmq::Error::EBUSY) => {
+                debug!("Socket Busy, skip");
+                std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
Is sleeping within this function a good idea? We're hiding a sleep within a very small function that can be used in any context, and incurring a sleep that someone may not realize they're paying for. Think we should ideally be sleeping in the surrounding context that this function is called so that we're not hiding this sleep when ever this fn is called.